app/models/piece.rb

```diff
@@ -11,14 +11,27 @@ def move_to!(new_row, new_col)
     check_if_castling(new_row, new_col) if type == "King"
     # Checking for Valid Move
     return unless valid_move?(new_row, new_col)
+    # Checking if the piece is moving into check
+    return unless !moving_into_check?(new_row, new_col)
```

Comment:
a minor thing, but I think we should change this to 
`return if moving_into_check?(new_row, new_col)` because it makes it slightly more readable :smile: 

In the Ruby Style Guide I think it mentioned this as well (avoid using `unless !some_method` and use `if some_method` instead)